src/components/ScenarioParameters/components/ScenarioParametersInputs/GenericToggleInput.js

```diff
@@ -7,36 +7,29 @@ import { useTranslation } from 'react-i18next';
 import PropTypes from 'prop-types';
 import { TranslationUtils } from '../../../../utils';
 
-export const GenericToggleInput = ({ parameterData, parametersState, setParametersState, context }) => {
+export const GenericToggleInput = ({ parameterData, context, parameterValue, setParameterValue }) => {
   const { t } = useTranslation();
   const switchFieldProps = {
     disabled: !context.editMode,
     id: parameterData.id,
   };
 
-  function setValue(newValue) {
-    setParametersState((currentParametersState) => ({
-      ...currentParametersState,
-      [parameterData.id]: newValue,
-    }));
-  }
-
   return (
     <BasicToggleInput
       key={parameterData.id}
       data-cy={`toggle-input-${parameterData.id}`}
       label={t(`solution.parameters.${parameterData.id}`, parameterData.id)}
       tooltipText={t(TranslationUtils.getParameterTooltipTranslationKey(parameterData.id), '')}
-      value={parametersState[parameterData.id] || false}
-      changeSwitchType={setValue}
+      value={parameterValue || false}
```

Comment:
Not part of the code you changed, but we could replace the `||` operator by `??` in all input components (especially for a bool value like this one)

Reply:
done